### PR TITLE
Bootstrap does not require xctool

### LINKED ISSUE #5: Bootstrap script includes xctool by default
The `cibuild` script requires `xctool` but the check done in `bootstrap` as a default requirement in its requirements checking step.

This has two drawbacks:
- `cibuild` can be used without `bootstrap`, and the check wouldn't be done in that case. If `xctool` were to be missing the build would fail without printing any error.
- `bootstrap` cannot be used without `xctool`, even though it has no need for `xctool`, and it's meant to be usable on its own, without `cibuild`.

### PR DESCRIPTION
**Pull request by robrix**
Fixes #5 such that:
- `cibuild` checks for its own dependencies (and thus is usable in the absence of `bootstrap`)
- `bootstrap` no longer depends on `xctool`

NB: I removed the `config` and `check_deps` portions of `bootstrap` entirely since they are now unused.

### CODE REVIEW

**Comment by robrix** [comment]
Motivation: make https://github.com/Carthage/Carthage/issues/19 easier.

File: cibuild [hunk 1]
```diff
@@ -78,6 +88,19 @@ main ()
     )
 }
 
+check_deps ()
+{
+    for tool in $REQUIRED_TOOLS
+    do
+        which -s "$tool"
+        if [ "$?" -ne "0" ]
+            then
```
Comment (jspahrsummers) [inline cibuild]: Can you match the indentation of `if` here?

**Comment by jspahrsummers** [comment]
Awesome! I've been wanting to do this for a long time, but it just never happened.

Note that [not all consumers](https://github.com/libgit2/objective-git/blob/9f433d2e85af674c9a22a95772bbe7e3e4f40ef0/script/bootstrap#L12) require _only_ `xctool`, so the removal of the `REQUIRED_TOOL` stuff might hamper them. But, they can easily reintroduce it into their own scripts (part of the virtue of subtree merging), so perhaps it's not a big deal.

/cc @Coneko

**Comment by robrix** [comment]
I’d be content with leaving config & the check_deps stuff in `bootstrap` but empty, too.

File: cibuild (hunk 1 above)
Comment (robrix) [inline cibuild]: :+1: good catch. copy pasta ahoy

**Comment by Coneko** [comment]
I completely forgot about this.

Yeah, looks good and solves all the problems I had.

**Pull request merged**